include/deal.II/lac/trilinos_solver.h

```diff
@@ -114,25 +114,19 @@ namespace TrilinosWrappers
        * base class for all solvers.
        */
       explicit
-      AdditionalData (const bool         output_solver_details   = false,
-                      const unsigned int gmres_restart_parameter = 30);
```

Comment:
But doing this breaks [this functionality](https://github.com/dealii/dealii/blob/master/include/deal.II/lac/trilinos_solver.h#L72) of using `SolverBase` as a "SolverSelector". I can imagine that this would cause both our unit tests and some user codes to fail.

As far as I can see, only other way to implement the necessary fix would be to revert to the original structure and to do the following:
```c++
  SolverCG::SolverCG (SolverControl        &cn,
                      const AdditionalData &data) // This is SolverCG's AdditionalData
    :
    SolverBase (cn,
                SolverBase::AdditionalData(data.output_solver_details)), // Pass down flags to base class. Its gmres_restart_parameter would take on a default value.
    additional_data (data.output_solver_details)
  {
    solver_name = cg;
  }
```
As discussed in our original conversation, I would consider this a suitable fix as well (its actually the "easier" of the two options).